discord/client.py

```diff
@@ -630,7 +630,12 @@ async def login(self, token: str) -> None:
         if self.loop is _loop:
             await self._async_setup_hook()
 
-        data = await self.http.static_login(token.strip())
+        try:
+            token = token.strip()
+        except AttributeError as exc:
+            raise TypeError('Token must be of type str.') from exc
```

Comment:
This isn't the proper way to do this. There are two ways of doing this:

The first is:

```suggestion
        if not isinstance(token, str):
            raise TypeError(f'expected token to be a str, received {token.__class__!r} instead')
        token = token.strip()
```

The second is:
```suggestion
        if token is None:
            raise TypeError('token must be a string')
        token = token.strip()
```

Reply:
^^